### PR TITLE
Extract tests to verify DELETE support

### PR DESCRIPTION
**Pull request by hashhar**
Follow-up to 9fb91ca1c7.

### CODE REVIEW

**Comment by hashhar** [comment]
Hopefully this is the last of these. I couldn't find any  more occurrences of `DELETE FROM nation` or `DELETE FROM region` other than Kudu tests (which are containerised so no issue).

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java [hunk 1]
```diff
@@ -1169,24 +1170,23 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
...
-        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey = 1", 1);
-        assertQuery(
-                "SELECT regionkey, name FROM " + tableName,
-                "VALUES "
-                        + "(0, 'AFRICA'),"
-                        + "(2, 'ASIA'),"
-                        + "(3, 'EUROPE'),"
-                        + "(4, 'MIDDLE EAST')");
-        assertUpdate("DROP TABLE " + tableName);
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA) && hasBehavior(SUPPORTS_ROW_LEVEL_DELETE));
```
Comment (losipiuk) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java:1173]: is `skipTestUnlessSupportsDeletes` still used?

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java [hunk 2]
```diff
@@ -1169,24 +1170,23 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
...
+                            + "(3, 'EUROPE'),"
+                            + "(4, 'MIDDLE EAST')");
+        }
     }
 
     @Test
     public void testDeleteWithVarcharEqualityPredicate()
     {
-        skipTestUnlessSupportsDeletes();
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_ROW_LEVEL_DELETE));
```
Comment (losipiuk) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java:1189]: nit: migration to `skipTestUnless` could be separate commit

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 3]
```diff
@@ -728,36 +730,51 @@ public void testRenameTableAcrossSchema()
     }
 
     @Test
-    public void testDeleteAllTable()
+    public void verifySupportsDelete()
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:733]: This name is misleading to me. Maybe `testDeclaredSupportsDeleteBehavior`?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 4]
```diff
@@ -728,36 +730,51 @@ public void testRenameTableAcrossSchema()
...
-        assertUpdate("DROP TABLE " + tableName);
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "AS SELECT * FROM nation")) {
+            assertQueryFails("DELETE FROM " + table.getName(), "This connector does not support deletes");
+        }
     }
 
     @Test
-    public void testRowLevelDelete()
+    public void verifySupportsRowLevelDelete()
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:747]: like above

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 5]
```diff
@@ -728,36 +730,51 @@ public void testRenameTableAcrossSchema()
...
-            return;
+
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "AS SELECT * FROM nation")) {
+            assertQueryFails("DELETE FROM " + table.getName() + " WHERE nationkey = 2", "This connector does not support deletes");
+        }
+    }
+
+    @Test
+    public void testDeleteAllTable()
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:761]: nit: testDeletaAllDataFromTable

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 6]
```diff
@@ -728,36 +730,51 @@ public void testRenameTableAcrossSchema()
...
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA) && hasBehavior(SUPPORTS_DELETE));
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:763]: Do we need `SUPPORTS_CREATE_TABLE_WITH_DATA`? I know test in this shape requires that, but I would rather have test fail for some connector which supports DELETE but not CREATE_TABLE_WITH_DATA; so we know we need to write it up differently for this connector, instead having silent skip and no testing for delete functinality.

**Review by losipiuk — COMMENTED** [review]
Thanks. Few stylistic suggestions and  https://github.com/trinodb/trino/pull/8505/commits/f5d99492549cef53b5c61fe01633c5c6c109b1d6#r666800843

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java (hunk 1 above)
Comment (hashhar) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java:1173]: Sadly for Kudu since it doesn't use BaseConnectorTest and uses ATDQ instead which doesn't have these flags. 😞

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 6 above)
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:763]: Fair point. Tests can be rewritten to use maybe CREATE+INSERT or some INSERT via remote database in subclasses.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 3 above)
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java]: Borrowed from BJCT - `verifySupports(RowLevel)DeleteDeclartion` (`verifySupportsJoinPushdownDeclaration`).

**Review by losipiuk — APPROVED** [review]
